### PR TITLE
chore(upgrade) upgrade script for 0.8.2 from 0.8.1

### PR DESCRIPTION
**Pull request by sonasingh46**
Upgrade script for 0.8.2 from 0.8.1
Signed-off-by: Ashutosh Kumar <ashutosh.kumar@openebs.io>

### CODE REVIEW

File: k8s/upgrades/0.8.1-0.8.2/pre-check.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+printf "\nCStor Target service:\n"
+echo "---------------------"
+printf "\n"
+# Search for CStor Target Service
+kubectl get service --all-namespaces -l 'openebs.io/version notin (0.8.2), openebs.io/version notin (0.8.1), openebs.io/target-service in (cstor-target-svc)'
+
+printf "\nCStor Target Deployment:\n"
+echo "---------------------"
+printf "\n"
+# Search for CStor Target Service
```
Comment (vishnuitta) [inline k8s/upgrades/0.8.1-0.8.2/pre-check.sh]: change the string as "Search for cStor Target deployment"

File: k8s/upgrades/0.8.1-0.8.2/cstor_pool_upgrade.sh [hunk 2]
```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+
+###########################################################################
+# STEP: Get SPC name and namespace where OpenEBS is deployed as arguments #
+#                                                                         #
+# NOTES: Obtain the pool deployments to perform upgrade operation         #
+###########################################################################
+
+pool_upgrade_version="v0.8.x-ci"
```
Comment (vishnuitta) [inline k8s/upgrades/0.8.1-0.8.2/cstor_pool_upgrade.sh]: this would be v0.8.2-ci

File: k8s/upgrades/0.8.1-0.8.2/cstor_pool_upgrade.sh [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+## Starting point
+if [ "$#" -ne 2 ]; then
+    usage
+fi
+
+spc=$1
+ns=$2
+
+### Get the deployment pods which are in not running state that are related to provided spc ###
+pending_pods=$(kubectl get po -n $ns \
```
Comment (vishnuitta) [inline k8s/upgrades/0.8.1-0.8.2/cstor_pool_upgrade.sh]: we don't need these checks

Comment (vishnuitta) [reply k8s/upgrades/0.8.1-0.8.2/cstor_pool_upgrade.sh]: and the next one also that looks at pending_pods

File: k8s/upgrades/0.8.1-0.8.2/cstor_volume_upgrade.sh [hunk 4]
```diff
@@ -0,0 +1,175 @@
+#!/bin/bash
+
+################################################################
+# STEP: Get Persistent Volume (PV) name as argument            #
+#                                                              #
+# NOTES: Obtain the pv to upgrade via "kubectl get pv"         #
+################################################################
+volume_upgrade_version="v0.8.x-ci"
```
Comment (vishnuitta) [inline k8s/upgrades/0.8.1-0.8.2/cstor_volume_upgrade.sh]: this should be v0.8.2-ci

File: k8s/upgrades/0.8.1-0.8.2/cstor_volume_upgrade.sh [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+fi
+
+### 1. Get the cstorvolume name related to the given PV ###
+### get the cloned volume cstorvolume name if exists
+echo "Upgrading Cstor Volume resource $volume_upgrade_version"
+cv_name=$(kubectl get cvr -n openebs\
+           -l openebs.io/persistent-volume=$pv\
+           -o 'jsonpath={.items[?(@.metadata.labels.openebs\.io/cloned=="true")].metadata.annotations.openebs\.io/source-volume}' | awk '{print $1}')
+
+version=$(verify_volume_version "cstorvolume" $pv)
```
Comment (vishnuitta) [inline k8s/upgrades/0.8.1-0.8.2/cstor_volume_upgrade.sh]: get cv from pv using command like `c_vol=$(kubectl get cstorvolumes -l openebs.io/persistent-volume=$pv -n $ns -o jsonpath="{.items[*].metadata.name}")`
don't assume pv is cv

File: k8s/upgrades/0.8.1-0.8.2/cstor_volume_upgrade.sh [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+# Check if CASType is cstor for PV
+cas_type=`kubectl get pv $pv -o jsonpath="{.metadata.labels.openebs\.io/cas-type}"`
+if [ $cas_type != "cstor" ]; then
+    echo "Cstor volume not found";exit 1;
+fi
+
+### 1. Get the cstorvolume name related to the given PV ###
+### get the cloned volume cstorvolume name if exists
+echo "Upgrading Cstor Volume resource $volume_upgrade_version"
```
Comment (vishnuitta) [inline k8s/upgrades/0.8.1-0.8.2/cstor_volume_upgrade.sh]: lets remove patching source-volume label in the cv

File: k8s/upgrades/0.8.1-0.8.2/cstor_volume_upgrade.sh [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+    rm cstor-volume-patch.json
+
+
+### 1. Get the cstorvolume name related to the given PV ###
+echo "Upgrading Target Service to $volume_upgrade_version"
+c_svc=$(kubectl get svc -n $ns\
+          -l openebs.io/persistent-volume=$pv,openebs.io/target-service=cstor-target-svc\
+          -o jsonpath="{.items[*].metadata.name}")
+
+version=$(verify_volume_version "service" $pv)
```
Comment (vishnuitta) [inline k8s/upgrades/0.8.1-0.8.2/cstor_volume_upgrade.sh]: use c_svc instead of pv

File: k8s/upgrades/0.8.1-0.8.2/cstor_volume_upgrade.sh [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+rc=$?
+if [ $rc -ne 0 ]; then
+   exit 1
+fi
+
+    ## 2. Update target svc patch file with upgrade version and pvc name
```
Comment (vishnuitta) [inline k8s/upgrades/0.8.1-0.8.2/cstor_volume_upgrade.sh]: fix the indentation in the script

File: k8s/upgrades/0.8.1-0.8.2/jiva_volume_upgrade.sh [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+}
+
+if [ "$#" -ne 1 ]; then
+    usage
+fi
+
+pv=$1
+replica_node_label="openebs-jiva"
+
+source snapshotdata_upgrade.sh
```
Comment (vishnuitta) [inline k8s/upgrades/0.8.1-0.8.2/jiva_volume_upgrade.sh]: lets pick snapshotdata_upgrade.sh file as well

Comment (vishnuitta) [reply k8s/upgrades/0.8.1-0.8.2/jiva_volume_upgrade.sh]: and, we need to add kn cstor_volume_upgrade script as well

File: k8s/upgrades/0.8.1-0.8.2/jiva_volume_upgrade.sh [hunk 10]
```diff
@@ -0,0 +1,227 @@
...
+    nl="";nl=`kubectl get nodes $rep_node -o jsonpath="{.metadata.labels.openebs-pv-$pv}"`
+    if [  -z "$nl" ];
+    then
+       echo "Labeling $rep_node";
+       kubectl label node $rep_node "openebs-pv-${pv}=$replica_node_label"
+    fi
+done
+
+
+sed -u "s/@r_name@/$r_name/g" | sed -u "s/@target_version@/$target_upgrade_version/g"  > jiva-replica-patch.json
```
Comment (vishnuitta) [inline k8s/upgrades/0.8.1-0.8.2/jiva_volume_upgrade.sh]: file name is missing.. also, @c_name@ is also missing

File: k8s/upgrades/0.8.1-0.8.2/jiva_volume_upgrade.sh [hunk 11]
```diff
@@ -0,0 +1,227 @@
+#!/bin/bash
+################################################################
+# STEP: Get Persistent Volume (PV) name as argument            #
+#                                                              #
+# NOTES: Obtain the pv to upgrade via "kubectl get pv"         #
+################################################################
+
+target_upgrade_version="0.8.2"
```
Comment (vishnuitta) [inline k8s/upgrades/0.8.1-0.8.2/jiva_volume_upgrade.sh]: this version would be v0.8.2-ci

File: k8s/upgrades/0.8.1-0.8.2/jiva_volume_upgrade.sh [hunk 12]
```diff
@@ -0,0 +1,227 @@
...
+if [[ "$controller_svc_version" != "$target_upgrade_version" ]]; then
+    echo "Upgrading Target Service to $target_upgrade_version"
+    kubectl patch service --namespace $ns $c_svc -p "$(cat jiva-target-svc-patch.json)"
+    rc=$?; if [ $rc -ne 0 ]; then echo "Failed to patch the service $svc | Exit code: $rc"; exit; fi
+else
+    echo "Controller service $c_svc is already at $target_upgrade_version"
+fi
+
+##Patch jiva snapshotdata crs related to pv
+run_snapshotdata_upgrades $pv
```
Comment (vishnuitta) [inline k8s/upgrades/0.8.1-0.8.2/jiva_volume_upgrade.sh]: no need of this

File: k8s/upgrades/0.8.1-0.8.2/cstor_volume_upgrade.sh (hunk 6 above)
Comment (sonasingh46) [reply k8s/upgrades/0.8.1-0.8.2/cstor_volume_upgrade.sh]: Done

File: k8s/upgrades/0.8.1-0.8.2/cstor_volume_upgrade.sh (hunk 4 above)
Comment (sonasingh46) [reply k8s/upgrades/0.8.1-0.8.2/cstor_volume_upgrade.sh]: Done

File: k8s/upgrades/0.8.1-0.8.2/cstor_pool_upgrade.sh (hunk 3 above)
Comment (sonasingh46) [reply k8s/upgrades/0.8.1-0.8.2/cstor_pool_upgrade.sh]: Done

File: k8s/upgrades/0.8.1-0.8.2/cstor_pool_upgrade.sh (hunk 2 above)
Comment (sonasingh46) [reply k8s/upgrades/0.8.1-0.8.2/cstor_pool_upgrade.sh]: Done

File: k8s/upgrades/0.8.1-0.8.2/pre-check.sh (hunk 1 above)
Comment (sonasingh46) [reply k8s/upgrades/0.8.1-0.8.2/pre-check.sh]: Done

File: k8s/upgrades/0.8.1-0.8.2/jiva_volume_upgrade.sh (hunk 11 above)
Comment (sonasingh46) [reply k8s/upgrades/0.8.1-0.8.2/jiva_volume_upgrade.sh]: Done

File: k8s/upgrades/0.8.1-0.8.2/jiva_volume_upgrade.sh (hunk 12 above)
Comment (sonasingh46) [reply k8s/upgrades/0.8.1-0.8.2/jiva_volume_upgrade.sh]: Done

File: k8s/upgrades/0.8.1-0.8.2/jiva_volume_upgrade.sh (hunk 10 above)
Comment (sonasingh46) [reply k8s/upgrades/0.8.1-0.8.2/jiva_volume_upgrade.sh]: Done

File: k8s/upgrades/0.8.1-0.8.2/jiva_volume_upgrade.sh (hunk 9 above)
Comment (sonasingh46) [reply k8s/upgrades/0.8.1-0.8.2/jiva_volume_upgrade.sh]: done

File: k8s/upgrades/0.8.1-0.8.2/cstor_volume_upgrade.sh [hunk 13]
```diff
@@ -5,8 +5,8 @@
 #                                                              #
 # NOTES: Obtain the pv to upgrade via "kubectl get pv"         #
 ################################################################
-volume_upgrade_version="v0.8.x-ci"
-volume_current_version="0.8.1"
+target_upgrade_version="0.8.2-RC4"
+current_version="0.8.0"
```
Comment (vishnuitta) [inline k8s/upgrades/0.8.1-0.8.2/cstor_volume_upgrade.sh]: current_version is 0.8.1

**Pull request merged**